erica/api/v2/endpoints/rente/fsc.py

```diff
@@ -0,0 +1,102 @@
+    except NotImplementedError:
+        logging.getLogger().info("Could not request unlock code", exc_info=True)
+        return JSONResponse(status_code=422, content=generate_dummy_error_response())
+
+
+@router.get('/request/{request_id}', status_code=status.HTTP_200_OK, response_model=ResponseGetFromQueue,
+            responses={500: {"model": ErrorRequestQueue}})
+@version(2)
+def get_request_fsc_job(request_id: str):
+    """
+    Route for retrieving job status from an fsc request from the queue.
+    :param request_id: the id of the job.
+    """
+    try:
+        raise NotImplementedError()
+    except NotImplementedError:
+        logging.getLogger().info("Could not retrieve status of (unlock code request) job " + request_id, exc_info=True)
+        return JSONResponse(status_code=500, content=generate_dummy_error_response())
+
+
+@router.post('/activation', status_code=status.HTTP_201_CREATED, responses={422: {"model": ErrorRequestQueue}})
+@version(2)
+def activation_fsc(
+        activation_fsc_ttl: FscActivationDataWithTtl):
+    """
+    Route for requesting activation of an fsc for the sent id_nr using the job queue.
+    :param activation_fsc_ttl: payload with TTL and the JSON input data for the activation.
+    """
+    try:
+        raise NotImplementedError()
+    except NotImplementedError:
+        logging.getLogger().info("Could not activate unlock code", exc_info=True)
+        return JSONResponse(status_code=422, content=generate_dummy_error_response())
+
+
+@router.get('/activation/{request_id}', status_code=status.HTTP_200_OK, response_model=ResponseGetFromQueue,
+            responses={500: {"model": ErrorRequestQueue}})
+@version(2)
+def get_fsc_activation_job(request_id: str):
+    """
+    Route for retrieving job status from an fsc activation from the queue.
+    :param request_id: the id of the job.
+    """
+    try:
+        raise NotImplementedError()
+    except NotImplementedError:
+        logging.getLogger().info("Could not retrieve status of (unlock code activation) job " + request_id,
+                                 exc_info=True)
+        return JSONResponse(status_code=500, content=generate_dummy_error_response())
+
+
+@router.post('/revocation', status_code=status.HTTP_201_CREATED, responses={422: {"model": ErrorRequestQueue}})
+@version(2)
+def fsc_revocation(
+        fsc_revocation_ttl: FscRevocationDataWithTtl):
+    """
+    Route for requesting revocation of an fsc for the sent id_nr using the job queue.
+    :param fsc_revocation_ttl: payload with TTL and the JSON input data for the revocation.
+    """
+    try:
+        raise NotImplementedError()
+    except NotImplementedError:
+        logging.getLogger().info("Could not revoke unlock code", exc_info=True)
+        return JSONResponse(status_code=422, content=generate_dummy_error_response())
+
+
+@router.get('/revocation/{request_id}', status_code=status.HTTP_200_OK, response_model=ResponseGetFromQueue,
```

Comment:
i was just wondering if we should use special response_models for the different responses that contains the actual data that will be send and not only the payload field. Wdyt?

Reply:
Yes that would be better! So I will refer to the response payloads of v1 for this, thanks!